src/types/decimal.rs

```diff
@@ -185,37 +206,69 @@ impl TryFrom<Decimal> for BigDecimal {
 
 #[cfg(test)]
 mod decimal_tests {
-    use crate::result::IonError;
+    use crate::result::IonResult;
     use crate::types::coefficient::{Coefficient, Sign};
     use crate::types::decimal::Decimal;
     use bigdecimal::BigDecimal;
     use num_traits::ToPrimitive;
+    use std::cmp::Ordering;
     use std::convert::TryInto;
+    use std::str::FromStr;
+
+    use rstest::*;
 
     #[test]
-    fn test_decimal_eq() {
```

Comment:
I replaced the monolithic `test_decimal_eq` and `test_decimal_ord` with `#[rstest]`-parameterized test functions.